ttl/ttlworker/job_manager.go

```diff
@@ -477,30 +477,34 @@ tblLoop:
 }
 
 // couldTrySchedule returns whether a table should be tried to run TTL
-func (m *JobManager) couldTrySchedule(table *cache.TableStatus, now time.Time) bool {
-	if table == nil {
+func (m *JobManager) couldTrySchedule(tableStatus *cache.TableStatus, table *cache.PhysicalTable, now time.Time) bool {
+	if tableStatus == nil {
 		// if the table status hasn't been created, return true
 		return true
 	}
-	if table.CurrentJobOwnerID != "" {
+	if table == nil {
+		// if the table is not recorded in info schema, return false
+		return false
+	}
+	if tableStatus.CurrentJobOwnerID != "" {
 		// see whether it's heart beat time is expired
-		hbTime := table.CurrentJobOwnerHBTime
+		hbTime := tableStatus.CurrentJobOwnerHBTime
 		// a more concrete value is `2 * max(updateTTLTableStatusCacheInterval, jobManagerLoopTickerInterval)`, but the
 		// `updateTTLTableStatusCacheInterval` is greater than `jobManagerLoopTickerInterval` in most cases.
 		if hbTime.Add(2 * updateTTLTableStatusCacheInterval).Before(now) {
-			logutil.Logger(m.ctx).Info("task heartbeat has stopped", zap.Int64("tableID", table.TableID), zap.Time("hbTime", hbTime), zap.Time("now", now))
+			logutil.Logger(m.ctx).Info("task heartbeat has stopped", zap.Int64("tableID", tableStatus.TableID), zap.Time("hbTime", hbTime), zap.Time("now", now))
 			return true
 		}
 		return false
 	}
 
-	if table.LastJobFinishTime.IsZero() {
+	if tableStatus.LastJobFinishTime.IsZero() {
 		return true
 	}
 
-	finishTime := table.LastJobFinishTime
+	finishTime := tableStatus.LastJobFinishTime
 
-	return finishTime.Add(variable.TTLJobRunInterval.Load()).Before(now)
+	return finishTime.Add(table.TTLInfo.CronJobSchedule.Duration()).Before(now)
```

Comment:
Could you confirm if this is the expected behavior? If we follow the [CRON syntax](https://cron.help/), that would mean that it starts 'at the start of each hour/day'.

Reply:
It's not a CRON syntax. Actually it only supports "@hourly" and "@daily", one returns `time.Hour`, the other returns `24 * time.Hour`.

I've discussed this behavior with @SunRunAway. The feedback is that we only need "@hourly" and "@daily", so don't provide more complex options for users.